ui/src/app/route-reuse.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  DetachedRouteHandle,
+  RouteReuseStrategy,
+  Routes
+} from '@angular/router';
+
+// Configure reuse of particular routes so components do not have to re-load
+@Injectable()
+export class RouteReuse implements RouteReuseStrategy {
+
+  public static cached: { [key: string]: any } = {};
```

Comment:
I'm wondering whether this cache would make more sense to put into the `JobListResolver`. Thoughts?

Reply:
I kind of like it here since this is the class which fetches and stores from the cache. Instead I added a public interface to this class for determining if a route is cached and getting the cached data. Let me know if you think that's better.

Reply:
OK - that makes sense. Could you make this not static and inject it into the JobListResolver instead then?